dll/win32/shell32/folders/CRecycleBin.cpp

```diff
@@ -983,14 +983,28 @@ TRASH_TrashFile(LPCWSTR wszPath)
 }
 
 /*************************************************************************
- * SHUpdateCRecycleBinIcon                                [SHELL32.@]
+ * SHUpdateRecycleBinIcon                                [SHELL32.@]
  *
  * Undocumented
  */
 EXTERN_C HRESULT WINAPI SHUpdateRecycleBinIcon(void)
```

Comment:
`SHUpdateRecycleBinIcon`'s logic of updating the trash bin icon is like so, in the following order:

- **Determine if the trash bin is empty or not**. It first iterates all the drive partitions and check each of them that one has the location path of the bucket that is associated as the default bucket bin of the system. Afterwards it loops the bucket of potential remaining files using the traditional fashion of `FindFirstFileW ... FindNextFileW` API calls. If the file in question is a directory (pointed by the **FILE_ATTRIBUTE_DIRECTORY** attribute) then it iterates over that directory as well. If at least one or more files have been found (empty directories are taken into account as well) then the bin is **NOT** empty.
- **Query/Set the Registry SHELL32 data**. Presumably it queries some information in regard of the trash bin icons and sets the default value as either "Empty" or "Full" depending on the state of the bin, I haven't looked much deeper into that.
- **Expand the environment strings with `SHExpandEnvironmentStringsW`**.
- **Update the icon image with `SHUpdateImageW`**.
- **Invoke a notification with `SHChangeNotify`**.

Now in the grand scheme of things I'm not sure if invoking a notification straight away alone in our case would be enough so ultimately the decision is up to @yagoulas.

Reply:
> empty directories are taken into account as well

How? Do only empty directories signal a full bin or an empty one?